File: Codility/003_02_PermMissingElem.cpp
// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    // write your code in C++14 (g++ 6.2.0)

    int n = A.size();
    int ans = 0;

    int sumI = 0;
    int sumA = 0;
    for(int i = 0; i < n; i++) {
        sumI += (i+1);
        sumA += A[i];
    }
    sumI += (n+1);

    return (sumI-sumA);
}

File: Codility/003_02_PermMissingElem.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int[] A) {
        // write your code in Java SE 8
        int N = A.length;

        long sum = 0;
        long finalSum = N+1;

        for(int i = 0; i < N; i++) {
            sum += A[i];
            finalSum += (i+1);
        }

        return (int)(finalSum-sum);
    }
}
